### DISCUSSION TITLE
Is there any way to use the request without controller

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1450] [Help] [answered]
right now I have a class called `Grid` and want to fetch the request without send the request to it like below

```javascript
class PostController extends Controller {
     
     async index({ request }) {
                 const query = Post.where('id', '>', 1);

                  const grid = await new Grid(query);
    }
}
```

I like the grid class work just with query and can access the request on that for sort and do some other query on that

```javascript

class Grid {
  constructor(query){
       this.query = query;
   }

    process() {
          // create query by request  and apply filters on it
          // apply sort by request 
    }

}

```

is there any way to use `request` without pass it from controller to the class

**Comment by iammrsea** [comment, marked as answer]
Your question is not clear enough, but lemme try pointing out few things.

1. You don't have access to the request object for a given request-response cycle in a plain class (like your Grid class) without explicitly passing the request object to it. So what you could do in your controller class is to pass the request object to Grid class.
```javascript
const grid = new Grid(query,request)
```

2. So, yeah there is no way you can use it without passing it from a class that has access to it (e.g contoller, middleware)
